lib/ansible/module_utils/common/validation.py

```diff
@@ -509,7 +509,7 @@ def check_type_int(value):
     if isinstance(value, integer_types):
         return value
 
-    if isinstance(value, string_types):
+    if isinstance(value, string_types) and value.lstrip('-').isdigit():
```

Comment:
What case(s) does the `isdigit` check handle that aren't already covered by the existing code?

Reply:
`0.1`

Reply:
`int('0.1')` raises a `ValueError`, so is already covered by this code, is it not?

Reply:
ah, was testing `int(0.1)` but that is not a string, in any case nitz's code with 'decimal' tracking seems a better fit.